### PR TITLE
Server/fecture/account hotel manager marge into develop

### PR DESCRIPTION
**Pull request by cangi75**
added server controler srevice and data access to endpoint /hotelInfo

### CODE REVIEW

File: Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        public HotelAccountServiceTest()
+        {
+            _dataAccessMock = new Mock<IHotelAccountDataAccess>();
+            hotelAccountService = new HotelAccountService(_dataAccessMock.Object);
+        }
+        [Fact]
+        public void GodGetingTest()
+        {
+            int hotelID = 1;
+            _dataAccessMock.Setup(x => x.AddHotelInfo(It.IsAny<HotelUpdateInfo>())).Returns(hotelID);
```
Comment (cyaneko) [inline Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs:28]: Is this even needed if you're not going to call the mock method later?

File: Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            
+            var h = hotelAccountService.GetInfo(hotelID);
+
+            Assert.True(h == hotel);
+        }
+        [Fact]
+        public void BadIDGetingTest()
+        {
+            int hotelID = 1;
+            _dataAccessMock.Setup(x => x.AddHotelInfo(It.IsAny<HotelUpdateInfo>())).Returns(hotelID);
```
Comment (cyaneko) [inline Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs:43]: Same as above

File: Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            
+            Assert.Throws<Exception>(() => hotelAccountService.GetInfo(hotelID + 1));
+            
+        }
+        
+        [Fact]
+        public void BadIDUpdateingTest()
+        {
+            int hotelID = 1;
+            _dataAccessMock.Setup(x => x.AddHotelInfo(It.IsAny<HotelUpdateInfo>())).Returns(hotelID);
```
Comment (cyaneko) [inline Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs:58]: Same as above

File: Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            _dataAccessMock.Setup(x => x.UpdateInfo(hotelID, hotel));
+
+            Assert.Throws<Exception>(() => hotelAccountService.UpdateInfo(hotelID + 1,hotel));
+
+        }
+        [Fact]
+        public void NullUpdateingTest()
+        {
+            int hotelID = 1;
+            _dataAccessMock.Setup(x => x.AddHotelInfo(It.IsAny<HotelUpdateInfo>())).Returns(hotelID);
```
Comment (cyaneko) [inline Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs:72]: Same as above

Comment (cangi75) [reply Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs]: done it

File: Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs (hunk 3 above)
Comment (cangi75) [reply Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs]: done it

File: Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs (hunk 2 above)
Comment (cangi75) [reply Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs]: done it

File: Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs (hunk 1 above)
Comment (cangi75) [reply Server-Module/Server.Tests/Services/HotelAccountServiceTest.cs]: done it

**Review by Michalski-Artur — CHANGES_REQUESTED** [review]
Server module not building. "The type or namespace name 'Exceptions' does not exist in the namespace 'Server'"

**Review by mtabor99-pw — APPROVED** [review]

**Review by cyaneko — APPROVED** [review]

File: Server-Module/Server/ViewModels/HotelUpdateInfo.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    public class HotelUpdateInfo
+    {
+        public string HotelName { get; set; }
+        public string HotelDesc { get; set; }
+        public string HotelPreviewPicture { get; set; }
+        public List<string> HotelPictures { get; set; }
+
+        public HotelUpdateInfo() { }
+
+        public HotelUpdateInfo(HotelInfoDb hotelInfoDb)
```
Comment (Michalski-Artur) [inline Server-Module/Server/ViewModels/HotelUpdateInfo.cs:19]: .

File: Server-Module/Server/Services/HotelAccountService/HotelAccountService.cs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                hotelAccountDataAccess.AddPictures(h);
+                hotelAccountDataAccess.UpdateInfo(h);
+            }catch(Exception e)
+            {
+                _transaction.RollbackTransaction();
+                return new ServiceResult(System.Net.HttpStatusCode.NotFound, new Error(e.Message));
+                
+            }
+            _transaction.CommitTransaction();
+            return new ServiceResult(System.Net.HttpStatusCode.OK, null);
```
Comment (Michalski-Artur) [inline Server-Module/Server/Services/HotelAccountService/HotelAccountService.cs:61]: .

**Review by Michalski-Artur — CHANGES_REQUESTED** [review]

File: Server-Module/Server/Database/DataAccess/HotelAccount/HotelAccountDataAccess.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            if (hotelInfoDb == null || hotelInfoDb.HotelPictures == null)
+                return;
+            dbContext.HotelPictures.AddRange(hotelInfoDb.HotelPictures);
+            dbContext.SaveChanges();
+        }
+        public void DeletePicteres(HotelInfoDb hotelInfoDb)
+        {
+            if (hotelInfoDb == null ||hotelInfoDb.HotelPictures==null)
+                return;
+            var pic = dbContext.HotelPictures.Find(hotelInfoDb.HotelPictures);
```
[the remark this answers is not in the document]
Comment (cangi75) [reply Server-Module/Server/Database/DataAccess/HotelAccount/HotelAccountDataAccess.cs:28]: >

File: Server-Module/Server/Database/DataAccess/HotelAccount/HotelAccountDataAccess.cs [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            dbContext.HotelPictures.RemoveRange(pic);
+            dbContext.SaveChanges();
+        }
+
+        public List<string> FindPictres(int hotelId)
+        {
+            var pom = dbContext.HotelPictures.AsEnumerable().Where(x => x.HotelID == hotelId);
```
[the remark this answers is not in the document]
Comment (cangi75) [reply Server-Module/Server/Database/DataAccess/HotelAccount/HotelAccountDataAccess.cs:35]: ?

File: Server-Module/Server/Database/DataAccess/HotelAccount/HotelAccountDataAccess.cs [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        private ServerDbContext dbContext;
+        public HotelAccountDataAccess(ServerDbContext dbContext)
+        {
+            this.dbContext = dbContext;
+        }
+        public void AddPictures(HotelInfoDb hotelInfoDb)
+        {
+            if (hotelInfoDb == null || hotelInfoDb.HotelPictures == null)
+                return;
+            dbContext.HotelPictures.AddRange(hotelInfoDb.HotelPictures);
```
[the remark this answers is not in the document]
Comment (cangi75) [reply Server-Module/Server/Database/DataAccess/HotelAccount/HotelAccountDataAccess.cs:21]: ?

File: Server-Module/Server/Services/HotelAccountService/HotelAccountService.cs [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+               
+            }
+            return new ServiceResult(System.Net.HttpStatusCode.OK, result);
+        }
+
+        public IServiceResult UpdateInfo(int hotelId, HotelUpdateInfo hotelUpdateInfo)
+        {
+            _transaction.BeginTransaction();
+            try {
+                var h = mapper.Map<HotelInfoDb>(hotelUpdateInfo);
```
[the remark this answers is not in the document]
Comment (cangi75) [reply Server-Module/Server/Services/HotelAccountService/HotelAccountService.cs:47]: ?

File: Server-Module/Server/ViewModels/HotelUpdateInfo.cs (hunk 5 above)
Comment (cangi75) [reply Server-Module/Server/ViewModels/HotelUpdateInfo.cs:19]: ?

File: Server-Module/Server/Services/HotelAccountService/HotelAccountService.cs (hunk 6 above)
Comment (cangi75) [reply Server-Module/Server/Services/HotelAccountService/HotelAccountService.cs:61]: ?

File: Server-Module/Server/Database/DataAccess/HotelAccount/HotelAccountDataAccess.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (Michalski-Artur) [reply Server-Module/Server/Database/DataAccess/HotelAccount/HotelAccountDataAccess.cs:28]: I would like to explain myself from dots (aka ".") tomorrow on discord

**Review by Michalski-Artur — APPROVED** [review]
Some refactoring will be still essential but I will hande it on my own tomorrow

**Pull request merged**